### PR TITLE
Davidr add type option for secrets

### PR DESCRIPTION
**Pull request by drohnow**
Draft pull request to add the option to set type of secret 
Ie: ability to set type: kubernetes.io/tls for secrets that contain tls data; if type is not set then will default to Opaque.

### CODE REVIEW

File: charts/replicated-library/templates/classes/_secret.tpl [hunk 1]
```diff
@@ -30,4 +30,5 @@ stringData:
 {{- with $values.data }}
   {{- tpl (toYaml .) $ | nindent 2 }}
 {{- end }}
+type: {{ $values.type}}
```
Comment (chris-sanders) [inline charts/replicated-library/templates/classes/_secret.tpl:33]:
```suggestion
type: {{ $values.type }}
```

Comment (chris-sanders) [reply charts/replicated-library/templates/classes/_secret.tpl:33]: Nitpick formatting, sometimes spaces around {{ actually matter. Here I don't know that it does just keeping it clean.

**Review by chris-sanders — CHANGES_REQUESTED** [review]
Well TIL: https://kubernetes.io/docs/concepts/configuration/secret/#secret-types
> You can define and use your own Secret type by assigning a non-empty string as the type value for a Secret object (an empty string is treated as an Opaque type).

> Kubernetes doesn't impose any constraints on the type name.

I was going to ask about validation but look at that, everything is valid. One small nit, I've included it as a recommendation so you can just accept it that's my only request.

File: charts/replicated-library/templates/classes/_secret.tpl (hunk 1 above)
Comment (drohnow) [reply charts/replicated-library/templates/classes/_secret.tpl]: shoot, I forgot I could just click "commit suggestion"...

**Review by chris-sanders — APPROVED** [review]
looks good now

**Pull request merged**